src/analysis/history.cpp

```diff
@@ -186,4 +186,57 @@ Real ReduceJetMomentumFlux(MeshData<Real> *md) {
 
 } // ReduceJetMomentumFlux
 
+Real ReduceMagneticFluxPhi(MeshData<Real> *md) {
+  const auto ib = md->GetBoundsI(IndexDomain::interior);
+  const auto jb = md->GetBoundsJ(IndexDomain::interior);
+  const auto kb = md->GetBoundsK(IndexDomain::interior);
+
+  auto pmb = md->GetParentPointer();
+  auto &pars = pmb->packages.Get("geometry")->AllParams();
+  const Real xh = pars.Get<Real>("xh");
+
+  namespace p = fluid_prim;
+  const std::vector<std::string> vars({p::bfield, p::velocity});
+
+  PackIndexMap imap;
+  auto pack = md->PackVariables(vars, imap);
+
+  const int pvel_lo = imap[p::velocity].first;
+  const int pvel_hi = imap[p::velocity].second;
+  const int pb_lo = imap[p::bfield].first;
+  const int pb_hi = imap[p::bfield].second;
+
+  auto geom = Geometry::GetCoordinateSystem(md);
+
+  Real result = 0.0;
+  parthenon::par_reduce(
+      parthenon::LoopPatternMDRange(), "Phoebus History for Jet Momentum Flux",
+      DevExecSpace(), 0, pack.GetDim(5) - 1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i, Real &lresult) {
+        const auto &coords = pack.GetCoords(b);
+        if (coords.x1f(i) <= xh && xh < coords.x1f(i + 1)) {
+          const Real dx1 = coords.Dx(X1DIR, k, j, i);
+          const Real dx2 = coords.Dx(X2DIR, k, j, i);
+          const Real dx3 = coords.Dx(X3DIR, k, j, i);
+
+          // interp to make sure we're getting the horizon correct
+          auto m = (CalcMagneticFluxPhi(pack, geom, pvel_lo, pvel_hi, pb_lo, pb_hi, b, k,
+                                        j, i + 1) -
+                    CalcMagneticFluxPhi(pack, geom, pvel_lo, pvel_hi, pb_lo, pb_hi, b, k,
+                                        j, i - 1)) /
+                   (2.0 * dx1);
+          auto flux = (CalcMagneticFluxPhi(pack, geom, pvel_lo, pvel_hi, pb_lo, pb_hi, b,
+                                           k, j, i) +
+                       (xh - coords.x1v(i)) * m) *
+                      dx2 * dx3;
+
+          lresult += flux;
+        } else {
+          lresult += 0.0;
+        }
+      },
+      result);
+  return 0.5 * result;
```

Comment:
Why is there a `0.5` here?

Reply:
Eq 14 for `\Phi_{BH}` in the EHT paper has it https://arxiv.org/pdf/1904.04923.pdf

Reply:
Ah ok. We might want to put that in the integrand then, rather than in the final integral. That might make things cleaner in the long run. But no big deal either way.